backend/src/app.ts

```diff
@@ -134,12 +134,15 @@ app.oauth = new OAuthServer({
 })
 
 const userCache = new UserCache(userRepository)
+// eslint-disable-next-line prefer-const
+let siteManager: SiteManager
+
 const notificationManager = new NotificationManager(
   commentRepository,
   notificationsRepository,
   postRepository,
-  siteRepository,
   userCache,
+  () => siteManager /*FIXME: HAX! circular dependency, will refactor later */,
```

Comment:
I know this is ugly, I have a plan to address dependencies once and for all in cleaner way, but:
1. it would blow up the scope of this PR
2. all other alternatives (like using `siteRepository` instead of `siteManager` are also ugly and will compromize performance)

considering this, I decided to leave this minor hack for now